From 3ed2c9ccd09745174f48d001cca8dc421d9e342a Mon Sep 17 00:00:00 2001
From: Sandeep <sandeepap@ami.com>
Date: Mon, 16 Oct 2023 15:07:02 +0530
Subject: [PATCH] FIXES LED button Display issue in Overview Page

Tested:

URI : https://{{IP}}/redfish/v1/Managers/bmc
Method : GET

Response :

{
"@odata.id": "/redfish/v1/Managers/bmc",
"@odata.type": "#Manager.v1_14_0.Manager",
"Actions": {
"#Manager.Reset": {
"@Redfish.ActionInfo": "/redfish/v1/Managers/bmc/ResetActionInfo",
"target": "/redfish/v1/Managers/bmc/Actions/Manager.Reset"
},
"#Manager.ResetToDefaults": {
"ResetType@Redfish.AllowableValues": [
"ResetAll",
"ResetToDefaultButKeepReservedSettings"
],
"target": "/redfish/v1/Managers/bmc/Actions/Manager.ResetToDefaults"
}
},
"CommandShell": {
"ConnectTypesSupported": [
"SSH",
"IPMI"
],
"MaxConcurrentSessions": 4,
"ServiceEnabled": true
},
"DateTime": "2023-10-16T09:36:11+00:00",
"DateTimeLocalOffset": "+00:00",
"Description": "Baseboard Management Controller",
"EthernetInterfaces": {
"@odata.id": "/redfish/v1/Managers/bmc/EthernetInterfaces"
},
"FirmwareVersion": "ot-0.3-100-ge00442-5d1fbc4",
"Id": "bmc",
"LastResetTime": "2023-10-16T09:26:20+00:00",
"Links": {
"ActiveSoftwareImage": {
"@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/bmc_active"
},
"ManagerForChassis": [
{
"@odata.id": "/redfish/v1/Chassis/AC_Baseboard"
}
],
"ManagerForChassis@odata.count": 1,
"ManagerForServers": [
{
"@odata.id": "/redfish/v1/Systems/system"
}
],
"ManagerForServers@odata.count": 1,
"ManagerInChassis": {
"@odata.id": "/redfish/v1/Chassis/AC_Baseboard"
},
"SoftwareImages": [
{
"@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/bmc_active"
}
],
"SoftwareImages@odata.count": 1
},
"LocationIndicatorActive": false,
"LogServices": {
"@odata.id": "/redfish/v1/Managers/bmc/LogServices"
},
"ManagerDiagnosticData": {
"@odata.id": "/redfish/v1/Managers/bmc/ManagerDiagnosticData"
},
"ManagerType": "BMC",
"Model": "OpenBmc",
"Name": "OpenBmc Manager",
"NetworkProtocol": {
"@odata.id": "/redfish/v1/Managers/bmc/NetworkProtocol"
},
"Oem": {
"@odata.id": "/redfish/v1/Managers/bmc#/Oem",
"@odata.type": "#OemManager.Oem",
"OpenBmc": {
"@odata.id": "/redfish/v1/Managers/bmc/Oem/OpenBmc/Jpeg",
"@odata.type": "#OemManager.OpenBmc",
"Certificates": {
"@odata.id": "/redfish/v1/Managers/bmc/Truststore/Certificates"
}
}
},
"PowerState": "On",
"SerialConsole": {
"ConnectTypesSupported": [
"IPMI",
"SSH"
],
"MaxConcurrentSessions": 15,
"ServiceEnabled": true
},
"ServiceEntryPointUUID": "df4322cb-e868-5b63-9e2e-b985fe0cadcf",
"Status": {
"Health": "OK",
"HealthRollup": "OK",
"State": "Enabled"
},
"UUID": "d021da1c-af15-4523-b472-42a7363e74f5",
"VirtualMedia": {
"@odata.id": "/redfish/v1/Managers/bmc/VirtualMedia"
}
}

Signed-off-by: Sandeep <sandeepap@ami.com>
---
 redfish-core/lib/managers.hpp | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 30fa0b01..00da6550 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -2056,6 +2056,7 @@ inline void requestRoutesManager(App& app)
                                              "FirmwareVersion", true);
 
         managerGetLastResetTime(asyncResp);
+        getLocationIndicatorActive(asyncResp);
 
         // ManagerDiagnosticData is added for all BMCs.
         nlohmann::json& managerDiagnosticData =
@@ -2230,9 +2231,11 @@ inline void requestRoutesManager(App& app)
         std::optional<nlohmann::json> oem;
         std::optional<nlohmann::json> links;
         std::optional<std::string> datetime;
+        std::optional<bool> locationIndicatorActive;
 
-        if (!json_util::readJsonPatch(req, asyncResp->res, "Oem", oem,
-                                      "DateTime", datetime, "Links", links))
+        if (!json_util::readJsonPatch(
+                req, asyncResp->res, "Oem", oem, "DateTime", datetime, "Links",
+                links, "LocationIndicatorActive", locationIndicatorActive))
         {
             return;
         }
@@ -2293,6 +2296,10 @@ inline void requestRoutesManager(App& app)
         {
             setDateTime(asyncResp, std::move(*datetime));
         }
+        if (locationIndicatorActive)
+        {
+            setLocationIndicatorActive(asyncResp, *locationIndicatorActive);
+        }
     });
 }
 
-- 
2.25.1

